examples/atlas-search.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+/**
+ * This example demonstrates how to create an Atlas Search index and perform a search query.
+ * It requires a MongoDB Atlas M10+ cluster with Sample Dataset loaded.
+ *
+ * Use the MONGODB_URI environment variable to specify the connection string from the Atlas UI.
+ */
+
+declare(strict_types=1);
+
+namespace MongoDB\Examples;
+
+use Closure;
+use MongoDB\Client;
+use RuntimeException;
+
+use function define;
+use function getenv;
+use function hrtime;
+use function iterator_to_array;
+use function sleep;
+use function str_contains;
+
+require __DIR__ . '/../vendor/autoload.php';
+
+$uri = getenv('MONGODB_URI');
+if (! $uri || ! str_contains($uri, '.mongodb.net')) {
+    echo 'This example requires a MongoDB Atlas cluster.', "\n";
+    echo 'Make sure you set the MONGODB_URI environment variable.', "\n";
+    exit(1);
+}
+
+// Atlas Search index management operations are asynchronous.
+// They usually take less than 5 minutes to complete.
+define('WAIT_TIMEOUT_SEC', 300);
+
+// The sample dataset is loaded into the "sample_airbnb.listingsAndReviews" collection.
+$databaseName = getenv('MONGODB_DATABASE') ?: 'sample_airbnb';
+$collectionName = getenv('MONGODB_COLLECTION') ?: 'listingsAndReviews';
+
+$client = new Client($uri);
+$collection = $client->selectCollection($databaseName, $collectionName);
+
+$count = $collection->estimatedDocumentCount();
+if ($count === 0) {
+    echo 'This example requires the "', $databaseName, '" database with the "', $collectionName, '" collection.', "\n";
+    echo 'Load the sample dataset in your MongoDB Atlas cluster before running this example:', "\n";
+    echo '    https://www.mongodb.com/docs/atlas/sample-data/', "\n";
+    exit(1);
+}
+
+// Delete the index if it already exists
+$indexes = iterator_to_array($collection->listSearchIndexes());
+foreach ($indexes as $index) {
+    if ($index->name === 'default') {
+        echo "The index already exists. Dropping it.\n";
+        $collection->dropSearchIndex($index->name);
```

Comment:
Since this is idempotent, I can remove surrounding loop.

Reply:
In fact no, the error for `NamespaceNotFound` is silenced, not the error for `IndexNotFound`. 
https://github.com/mongodb/specifications/blob/master/source/index-management/index-management.rst#namespacenotfound-errors